core/src/main/java/com/linecorp/armeria/server/RouteCache.java

```diff
@@ -108,14 +144,34 @@ private RouteCache() {}
 
             final Routed<V> result = delegate.find(routingCtx);
             if (result.isPresent()) {
-                cache.put(routingCtx, result.value());
+                findCache.put(routingCtx, result.value());
+            }
+            return result;
+        }
+
+        @Override
+        public List<Routed<V>> findAll(RoutingContext routingCtx) {
+            final List<V> cachedList = findAllCache.getIfPresent(routingCtx);
+            if (cachedList != null) {
+                return cachedList.stream().map(cached -> {
+                    final Route route = routeResolver.apply(cached);
+                    final RoutingResult routingResult = route.apply(routingCtx);
+                    return Routed.of(route, routingResult, cached);
+                }).collect(toImmutableList());
             }
+
+            final List<Routed<V>> result = delegate.findAll(routingCtx);
+            final List<V> valid = result.stream()
```

Comment:
What do you think about making `findAll` return a `Stream` instead of `List`, to avoid double collection?

Reply:
Now, `findAll` returns `Stream` but I have to collect two times to filter and store them into the cache. There is a good solution here, let me know :-)